tests/dockerConfiguration/publicAddressVariable_test.js

```diff
@@ -0,0 +1,109 @@
+  await I.verifyCommand('docker stop pmm-server-public-address');
+  await I.verifyCommand('docker rm pmm-server-public-address');
+});
+
+Data(publicIPs).Scenario(
+  'PMM-T1173 PMM-T1174 Verify PMM_PUBLIC_ADDRESS env variable with IP @docker-configuration',
+  async ({
+    I, pmmSettingsPage, current, perconaPlatformPage,
+  }) => {
+    const basePmmUrl = 'http://127.0.0.1:8085/';
+    const { publicAddress } = current;
+
+    await runContainerWithPublicAddressVariable(I, publicAddress);
+    await I.amOnPage(basePmmUrl + pmmSettingsPage.advancedSettingsUrl);
+    await I.waitForVisible(pmmSettingsPage.fields.publicAddressInput, 30);
+    const setPublicAddress = await I.grabValueFrom(pmmSettingsPage.fields.publicAddressInput);
+
+    assert.ok(setPublicAddress === publicAddress, 'Set public address does not equal to one specified in public address environment variable');
+
+    await I.amOnPage(basePmmUrl + perconaPlatformPage.url);
+    I.waitForVisible(perconaPlatformPage.elements.settingsContent, 30);
+    await perconaPlatformPage.connectToPortal(adminToken);
+  },
+);
+
+Scenario(
+  'PMM-T1176 Verify PMM_PUBLIC_ADDRESS env variable after upgrade @docker-configuration',
+  async ({
+    I, pmmSettingsPage, homePage,
+  }) => {
+    const basePmmUrl = 'http://127.0.0.1:8085/';
+
+    await runContainerWithPublicAddressVariableUpgrade(I, '127.0.0.1');
+    await I.amOnPage(basePmmUrl + homePage.url);
+    await I.waitForElement(homePage.fields.dashboardHeaderLocator, 60);
+    const { versionMinor } = await homePage.getVersions();
+
+    await homePage.upgradePMM(versionMinor);
+
+    await I.amOnPage(basePmmUrl + pmmSettingsPage.advancedSettingsUrl);
+    await I.waitForVisible(pmmSettingsPage.fields.publicAddressInput, 30);
+    const setPublicAddress = await I.grabValueFrom(pmmSettingsPage.fields.publicAddressInput);
+
+    assert.ok(setPublicAddress === '127.0.0.1', 'Set public address does not equal to one specified in public address environment variable');
+  },
+);
+
+Scenario(
+  'PMM-T117 Verify PMM_PUBLIC_ADDRESS env variable can be updated @docker-configuration',
+  async ({
+    I, pmmSettingsPage, perconaPlatformPage,
+  }) => {
+    const basePmmUrl = 'http://127.0.0.1:8085/';
+
+    await runContainerWithPublicAddressVariable(I, '127.0.0.5');
+    await I.amOnPage(basePmmUrl + pmmSettingsPage.advancedSettingsUrl);
+    await I.waitForVisible(pmmSettingsPage.fields.publicAddressInput, 30);
+    await pmmSettingsPage.clearPublicAddress();
+    await pmmSettingsPage.addPublicAddress('127.0.0.1');
+    const setPublicAddress = await I.grabValueFrom(pmmSettingsPage.fields.publicAddressInput);
+
+    assert.ok(setPublicAddress === '127.0.0.1', 'Set public address does not equal to one specified in public address environment variable');
+    await I.amOnPage(basePmmUrl + perconaPlatformPage.url);
+    I.waitForVisible(perconaPlatformPage.elements.settingsContent, 30);
+    await perconaPlatformPage.connectToPortal(adminToken);
```

Comment:
Any specific need for having Portal involved into this test case?

Reply:
test case asks to verify portal connection, but if you think that is enough that address is set I can remove it.